nixos/tests/all-tests.nix

```diff
@@ -77,7 +98,7 @@ in {
 
   _3proxy = runTest ./3proxy.nix;
   aaaaxy = runTest ./aaaaxy.nix;
-  acme = runTest ./acme.nix;
+  acme = runTest { imports = [ ./acme.nix readOnlyPkgs ]; };
```

Comment:
Is the plan to set this for all tests?

Reply:
I think that would be best, but I can't speak for everyone.
Most tests aren't in `runTest` style yet, and I don't know if everyone wants to switch their test to it.
Doing it for `handleTest` is too complicated for me. Too many callbacks and alternate ways to invoke stuff.

Reply:
We can default it for runTest for now, right?

Reply:
Some modules will set something in `nixpkgs.config`, or add an overlay. Wouldn't recommend, but that means that we can't apply this everywhere. I don't mind if "leaky" things like this are repeated. It's good to be reminded of it.

Reply:
I've added a `node.pkgsReadOnly` option to the test framework so that the optimization can be disabled if needed.
We could run it through the `python-test-refactoring` job on hydra.nixos.org to figure out which tests need an opt out. https://nixos.org/manual/nixos/unstable/index.html#sec-test-the-test-framework
Although that's not necessary for this PR.